src/app/child-dev-project/attendance/add-day-attendance/roll-call/roll-call.component.ts

```diff
@@ -64,9 +76,14 @@ export class RollCallComponent implements OnInit {
     private confirmationDialog: ConfirmationDialogService
   ) {}
 
-  async ngOnInit() {
-    this.loadAttendanceStatusTypes();
-    await this.loadParticipants();
+  async ngOnChanges(changes: SimpleChanges) {
+    if (changes.eventEntity) {
+      this.loadAttendanceStatusTypes();
+      await this.loadParticipants();
+    }
+    if (changes.sortParticipantsBy) {
+      this.sortParticipants();
+    }
```

Comment:
```suggestion
    this.sortParticipants();
```

If there is the check on line 119 I don't think there is a second check needed here

Reply:
it's not necessary to trigger a new sorting though, if that input did not change (especially if any completely unrelated inputs may be added in the future).

I included a sort in the loadParticipants now, however, to ensure this reflects the current sort state if new data comes in